lsp-mode.el

```diff
@@ -5592,36 +5587,58 @@ SESSION is the active session."
       (lsp--start-workspace session client project-root (lsp--create-initialization-options session client))
     (lsp--spinner-stop)))
 
+;; lsp-log-io-mode
+
+(defvar lsp-log-io-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "M-n") #'lsp--log-io-next)
+    (define-key map (kbd "M-p") #'lsp--log-io-prev)
+    (define-key map (kbd "k") #'lsp--erase-log-buffer)
+    map)
+  "Keymap for lsp log buffer mode.")
+
+(define-derived-mode lsp-log-io-mode special-mode "LspLogIo"
+  "Special mode for viewing IO logs.")
+
 (defun lsp-workspace-show-log (workspace)
+  "Display the log buffer of WORKSPACE."
   (interactive
    (list (if lsp-print-io
              (if (eq (length (lsp-workspaces)) 1)
                  (cl-first (lsp-workspaces))
                (lsp--completing-read "Workspace: " (lsp-workspaces)
                                      #'lsp--workspace-print nil t))
            (user-error "IO logging is disabled"))))
-  (switch-to-buffer (lsp--generate-log-buffer-name workspace)))
+  (switch-to-buffer (lsp--get-log-buffer-create workspace)))
 
 (defalias 'lsp-switch-to-io-log-buffer 'lsp-workspace-show-log)
 
-(defun lsp-log-io-next (arg)
+(defun lsp--get-log-buffer-create (workspace)
+  "Return the lsp log buffer of WORKSPACE, creating a new one if needed."
+  (let ((server-id (-> workspace lsp--workspace-client lsp--client-server-id symbol-name))
+        (pid (format "%s" (process-id (lsp--workspace-cmd-proc workspace)))))
+    (get-buffer-create (format "*lsp-log: %s:%s*" server-id pid))))
+
+(defun lsp--erase-log-buffer ()
+  "Delete contents of the project's lsp log buffers."
+  (interactive)
+  (let* ((session (lsp-session))
+         (workspaces (lsp--session-workspaces (lsp-session)))
+         (inhibit-read-only t))
+    (dolist (w workspaces)
+      (with-current-buffer (lsp--get-log-buffer-create w)
+        (erase-buffer)))))
+
+(defun lsp--log-io-next (arg)
+  "Move to next log entry."
   (interactive "P")
   (ewoc-goto-next lsp--log-io-ewoc (or arg 1)))
 
-(defun lsp-log-io-prev (arg)
+(defun lsp--log-io-prev (arg)
```

Comment:
Why do you rename the function? We previously decided to make this public so a user could eventually re-map them.

Reply:
Agree. Don't change the public APIs.